doc/dev/custom_devices.rst

```diff
@@ -181,3 +186,145 @@ After doing so, Catalyst should be able to interface with your custom device.
+                "IsingYY",
+                "IsingZZ",
+                "IsingXY",
+        ]
+
+        # Operators that should be decomposed according to the algorithm used
+        # by PennyLane's device API.
+        # Optional, since gates not listed in this list will typically be decomposed by
+        # default, but can be useful to express a deviation from this device's regular
+        # strategy in PennyLane.
+        decomp = [
+                "SX",
+                "ISWAP",
+                "PSWAP",
+                "SISWAP",
+                "SQISW",
+                "CPhase",
+                "BasisState",
+                "QubitStateVector",
+                "StatePrep",
+                "ControlledQubitUnitary",
+                "DiagonalQubitUnitary",
+                "SingleExcitation",
+                "SingleExcitationPlus",
+                "SingleExcitationMinus",
+                "DoubleExcitation",
+                "DoubleExcitationPlus",
+                "DoubleExcitationMinus",
+                "QubitCarry",
+                "QubitSum",
+                "OrbitalRotation",
+                "QFT",
+                "ECR",
+        ]
+        # Gates which should be translated to QubitUnitary
+        matrix = [
+                "MultiControlledX",
+        ]
+
+        [measurement_processes]
+        exactshots = [
+                "Expval",
+                "Var",
+                "Probs",
+                "State",
+        ]
+        finiteshots = [
+                "Expval",
+                "Var",
+                "Probs",
+                "Sample",
+                "Counts",
+        ]
+
+        [compilation]
+        # If the device is compatible with qjit
+        qjit_compatible = true
+        # If the device requires run time generation of the quantum circuit.
+        runtime_code_generation = false
+        # If the device supports adjoint
+        quantum_adjoint = true
+        # If the device supports quantum control instructions natively
+        quantum_control = false
+        # If the device supports mid circuit measurements natively
+        mid_circuit_measurement = true
+
+        # This field is currently unchecked but it is reserved for the purpose of
+        # determining if the device supports dynamic qubit allocation/deallocation.
+        dynamic_qubit_management = false 
+
+        [options]
+        # Options is an optional field.
+        # These options represent runtime parameters that can be passed to the device
+        # upon the device initialization.
+        # The option key will be the key in a dictionary.
+        # The string corresponds to a field queried in the `qml.Device` instance.
+        option = "option"
+        # The concrete meaning is as follows:
+        # if "options" in config.keys():
+        #    for k, v in config["options"].items():
+        #         if hasattr(device, v):
+        #             runtime_options[k] = getattr(device, v)
```

Comment:
Not blocking for this PR @erick-xanadu, just a question I had :) I recall that you mentioned that it would be better to encode this in the Python version of the device class (the options, default values, and value types), rather than in the TOML file

Reply:
Yes. @dime10 mentioned that it would be a good idea to have the options in the toml file along with the comment description. I do think that having the toml file specifying the options is not ideal. The method described above is limited to obtaining fields in the device object. Allowing the user to specify options that should be queried from the device's nested fields (or some other way) is likely not a good idea. Happy to make changes here.

We can also keep this description and bump the schema + 1 once we implemented the changes in `qml.device`.

Reply:
Yes this is an intermediate solution that has the benefit that we at least don't have to hardcode such options into the Catalyst frontend for every single device (which is not extensible).

Agreed that we can change it once `qml.Device` has a standard way to store device options, but I figured that would be a longer process given that different parties are involved.

Reply:
I'll close this comment, feel free to reopen if you would like something changed.